src/components/Popup/OfficePopup.vue

```diff
@@ -0,0 +1,96 @@
+        private originalFocusedElement?: HTMLElement;
+
+        private focused: boolean = false;
+
+        get classNames() {
+            return mergeStyleSets({
+                root: [
+                    this.needsVerticalScrollBar && {
+                        overflowY: "scroll"
+                    },
+                    !this.needsVerticalScrollBar && {
+                        overflowY: undefined
+                    }]
+            });
+        }
+
+        public beforeMount() {
+            this.originalFocusedElement = getDocument()!.activeElement as HTMLElement;
+        }
+
+        public mounted() {
+            this.updateScrollBar();
+        }
+
+        public updated() {
+            this.updateScrollBar();
+        }
+
+        public beforeDestroy() {
+            if (this.shouldRestoreFocus &&
+                this.originalFocusedElement &&
+                this.focused &&
+                (this.originalFocusedElement as any) !== window &&
+                this.originalFocusedElement) {
+                this.originalFocusedElement.focus();
+            }
+        }
+
+        public onSelfBlur(event: any) {
+            if ((this.$refs.self as HTMLElement).contains(event.relatedTarget as HTMLElement)) {
+                this.focused = false;
+            }
+        }
+
+        private updateScrollBar() {
+            requestAnimationFrame(() => this.getScrollBar());
+        }
+
+        private getScrollBar() {
+            if (this.needsVerticalScrollBar) {
+                return;
+            }
+            let needsVerticalScrollBar = false;
+            const root = this.$refs.self as HTMLElement;
+            if (root && root.firstElementChild) {
+                // ClientHeight returns the client height of an element rounded to an
+                // integer. On some browsers at different zoom levels this rounding
+                // can generate different results for the root container and child even
+                // though they are the same height. This causes us to show a scroll bar
+                // when not needed. Ideally we would use BoundingClientRect().height
+                // instead however seems that the API is 90% slower than using ClientHeight.
+                // Therefore instead we will calculate the difference between heights and
+                // allow for a 1px difference to still be considered ok and not show the
+                // scroll bar.
+                const rootHeight = root.clientHeight;
+                const firstChildHeight = root.firstElementChild.clientHeight;
+                if (rootHeight > 0 && firstChildHeight > rootHeight) {
+                    needsVerticalScrollBar = firstChildHeight - rootHeight > 1;
+                }
+            }
+            if (this.needsVerticalScrollBar !== needsVerticalScrollBar) {
+                this.needsVerticalScrollBar = needsVerticalScrollBar;
```

Comment:
This will overwrite a the prop "needsVerticalScrollBar", so we cannot use the variable here. Instead you probably have to define another state variable with a similar name.

Reply:
Nope, didn't resolve the issue. You are mutating `this.needsVerticalScrollBar`, which is a prop (`@Prop(...) private needsVerticalScrollBar`). You are not allowed to mutate props directly, as this change will be overwritten by the next render of the parent element. In the React implementation they have `props.needsVerticalScrollBar` and `state.needsVerticalScrollBar`. You don't have that differentiation currently